### DISCUSSION TITLE
Access camera locations (X,Y,Z)

### DISCUSSION THREAD

**Discussion by informatter** [alicevision/Meshroom#1737] [Q&A] [answered]
Hi everyone,

Is there a way I can get the camera locations (x,y,z)? I currently need to have access to the locations of the cameras used within Meshroom, It would also be interesting to see, where in the code the camera's locations get generated from each of the  image's lat long within the EXIF data, to Meshroom's internal coordinate system.

Thanks

Nicholas

**Comment by simogasp** [comment, marked as answer]
You should convert the sfmdata file from .abc (default format) to the json format. There you will find all the information about the scene, from the camera poses to the 3D points etc.
You should use the ConvertSfMFormat
https://meshroom-manual.readthedocs.io/en/bibtex1/node-reference/nodes/ConvertSfMFormat.html

**Comment by informatter** [comment]
@simogasp 
Thank you very much for the link. Also, I am curious to know in what part of AliceVision's src does each camera's position get computed?

**Reply by simogasp** [reply]
In the incremental sfm pipeline this is the part that iteratively compute the pose for each new image that shares some matches with those already reconstructed
https://github.com/alicevision/AliceVision/blob/develop/src/aliceVision/sfm/pipeline/sequential/ReconstructionEngine_sequentialSfM.cpp#L515

**Reply by informatter** [reply]
Thanks!
